subsys/bluetooth/mesh/prov.c

```diff
@@ -902,8 +894,11 @@ static void send_pub_key(const u8_t dhkey[32])
 		return;
 	}
 
-	atomic_set_bit(link.flags, SEND_CONFIRM);
-	link.expect = PROV_CONFIRM;
+	if (atomic_test_bit(link.flags, WAIT_NUMBER) || atomic_test_bit(link.flags, WAIT_STRING)) {
+		link.expect = 0xff; /* Don't expect any packets until after input */
```

Comment:
I just realised that we're being inconsistent with this:
https://github.com/zephyrproject-rtos/zephyr/blob/6abbed2975a0a3400658819a051e2ae81700c342/subsys/bluetooth/mesh/prov.c#L1099-L1100

0 is actually a bad value since it's a valid PDU type, whereas 0xff is indeed a good choice. It'd be good to fix up the existing 0 as well (separate patch though, can be this same PR if you want), but then please create an intuitive define for this, e.g. `#define PROV_NO_PDU 0xff`